tests/microos/toolbox.pm

```diff
@@ -0,0 +1,85 @@
+# SUSE's openQA tests
+#
+# Copyright © 2021 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Run simple toolbox tests
+# Maintainer: Jose Lausuch <jalausuch@suse.com>
+
+use base "opensusebasetest";
+use strict;
+use warnings;
+use testapi;
+use version_utils 'is_sle_micro';
+
+
+sub run {
+    select_console 'root-console';
+    my $user = 'test_user';
+
+    # Create user and don't ask password for sudo commands
+    assert_script_run "useradd -m $user ";
+    assert_script_run "echo \"$user ALL=(ALL) NOPASSWD:ALL\" >> /etc/sudoers";
+
+    # Display help
+    record_soft_failure 'https://github.com/kubic-project/microos-toolbox/issues/22' if (script_run('toolbox -h') != 0);
+
+    record_info 'Test',                    "Run toolbox without flags";
+    assert_script_run 'toolbox id',        timeout => 180;
+    validate_script_output 'podman ps -a', sub { m/toolbox-root/ };
+    assert_script_run 'podman rm toolbox-root';
+
+    record_info 'Test',                         "Run toolbox with a given tag";
+    assert_script_run 'toolbox -t test_tag id', timeout => 180;
+    validate_script_output 'podman ps -a',      sub { m/toolbox-root-test_tag/ };
+    assert_script_run 'podman rm toolbox-root-test_tag';
+
+    record_info 'Test', "Run toolbox with a given name";
+    if (script_run('toolbox -c test_name id') == 0) {
+        validate_script_output 'podman ps -a', sub { m/test_name/ };
+        assert_script_run 'podman rm test_name';
+    } else {
+        record_soft_failure 'https://github.com/kubic-project/microos-toolbox/issues/19';
+    }
+
+    record_info 'Test', "Rootless toolbox as $user";
+    my $prefix = "runuser -l $user -c";
+    my $uid    = script_output "$prefix 'id -u'";
+    validate_script_output "$prefix 'toolbox -u id'", sub { m/uid=${uid}\(${user}\)/ }, timeout => 180;
+    die "$user shouldn't have access to /etc/passwd!" if (script_run("$prefix 'toolbox -u touch /etc/passwd'") == 0);
+    assert_script_run "$prefix \"podman rm toolbox-$user-user\"";
+
+    record_info 'Test',                               "Rootfull toolbox as $user";
+    validate_script_output "$prefix 'toolbox -r id'", sub { m/uid=0\(root\)/ };
+    die "Root should have access to /etc/passwd!" if (script_run("$prefix 'toolbox -r touch /etc/passwd'") != 0);
```

Comment:
Why not just `assert_script_run`?

Reply:
Just to make clear the reason of the failure with that message.

Reply:
Why this command in particular and not all other `assert_script_run`s?

Reply:
If I use `assert_script_run` and the command fails, we won't know the reason. By using `die` + `script_run` we make sure that if it fails, we know why. If you think that the reason is too obvious I can change it to `assert_script_run`

Reply:
You could put that into the `record_info` above or as a comment. IMO it's obvious enough already - the command doesn't really do anything except prove that there's write access.

Reply:
> If I use `assert_script_run` and the command fails, we won't know the reason. By using `die` + `script_run` we make sure that if it fails, we know why. If you think that the reason is too obvious I can change it to `assert_script_run`

https://github.com/os-autoinst/os-autoinst/blob/master/testapi.pm#L965 it has the `fail_message` key for such a thing. But that is just sugar syntax :)

Reply:
thanks for the hint, I updated it.